sunpy/util/scraper.py

```diff
@@ -128,6 +153,18 @@ def _URL_followsPattern(self, url):
         """
         Check whether the url provided follows the pattern.
         """
+        if self.matcher is not None:
```

Comment:
matches using the `**kwargs` of client query. It checks whether the info extracted from `url` have attrs one among all possible attrs.